clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -1111,6 +1112,11 @@ scanl f z xs = ws
     ws = z `Cons` zipWith (flip f) xs (init ws)
 {-# INLINE scanl #-}
 
+-- | 'scanl' with no seed value
```

Comment:
I would improve this documentation. It doesn't need to be as complete as the ones with big descriptions and diagrams, but should have an example in a similar style to `zip`, i.e.

```
-- | 'zip' takes two vectors and returns a vector of corresponding pairs.
--
-- >>> zip (1:>2:>3:>4:>Nil) (4:>3:>2:>1:>Nil)
-- (1,4) :> (2,3) :> (3,2) :> (4,1) :> Nil
```

Reply:
This way we at least have a `doctest` for the function as a very basic sanity-check

Reply:
Yes! I will add doctests.